lib/LLVMIRCodeGen/AllocationsInfo.cpp

```diff
@@ -35,6 +35,45 @@ using llvm::cast;
 using llvm::dyn_cast;
 using llvm::isa;
 
+/// If \p PH is an output placeholder, \returns true.
+/// This is determined by checking if the PH has a user which uses the PH as an
+/// overwritten input.
+bool isOutputPlaceholder(const Placeholder *PH) {
+  for (const auto &use : PH->getUsers()) {
+    // Look through the inputs of the PH's users. If an input is overwritten
+    // check if it's the PH, if it is return true.
+    auto *user = use.getUser();
```

Comment:
Your function operates at the graph level, but you use it from a function that operates at the `IRFunction` level. This is wrong.

Also, you only want to consider uses in the current function, not in the whole module.

Reply:
> Your function operates at the graph level, but you use it from a function that operates at the `IRFunction` level. This is wrong.
> 
> Also, you only want to consider uses in the current function, not in the whole module.

This was just a copy of existing `isInput` and `isOutput` in the `BackendUtils.cpp`. (I already merged them into a helper function currently) I noticed that these two functions are called during the existing allocations and saved inside the `runtime::RuntimeSymbolInfo`.  I was not sure if there will be an issue if I change the logic here. Do you have any suggestions about the logic of the implementations ? :)